### ISSUE TITLE
Future desirable features

### ISSUE THREAD

**Issue by Alan32Liu** [google/fuzzbench#1590]
To keep track of the desirable features that we'd like to add in the future, here is a list of them in the order of priority:
1. [ ] [Make CI tests OSS-Fuzz-like trial builds](https://github.com/google/fuzzbench/issues/1556).
2. [ ] Revive more bug-based benchmarks and test them thoroughly with core fuzzers.
3. [ ] Allow users to add fuzzer flags from the command line or configuration file.
4. [ ] Versionize FuzzBench: We use a stable version of FuzzBench for production (experiments, etc.) and periodically replace it with new versions.
5. [ ] Update benchmarks yearly to ensure the timeliness of FuzzBench.

**Comment by Alan32Liu** [comment]
Happy to discuss the necessity/priority of each, feel free to add more if I missed any : )

**Comment by jonathanmetzman** [comment]
>  Allow users to add fuzzer flags from the command line or configuration file.

No I don't think this is worth it. I'd rather people make new fuzzers. We tried this and it's more trouble to implement than is worth it. Keep it Simple S*****

**Comment by jonathanmetzman** [comment]
Another feature: something to write all stackdriver logs from an experiment to a bucket that can be viewed by users so we can shift some debugging burden

**Comment by Alan32Liu** [comment]
> Another feature: something to write all stackdriver logs from an experiment to a bucket that can be viewed by users so we can shift some debugging burden

Yep I agree this could be very helpful, let me add it to the list.

**Comment by Alan32Liu** [comment]
> > Allow users to add fuzzer flags from the command line or configuration file.
> 
> No I don't think this is worth it. I'd rather people make new fuzzers. We tried this and it's more trouble to implement than is worth it. Keep it Simple S*****

Could this be as simple as passing a JSON file (e.g., in format `{libfuzzer: '--flag_a --flag b'}`) as an argument of `run_experiment.py`, or having extra items in the `experiment-config.yaml` (e.g., in format `extra_flags: -libfuzzer: '--flag_a --flag b'`)?

**Comment by jonathanmetzman** [comment]
> > > Allow users to add fuzzer flags from the command line or configuration file.
> > 
> > 
> > No I don't think this is worth it. I'd rather people make new fuzzers. We tried this and it's more trouble to implement than is worth it. Keep it Simple S*****
> 
> Could this be as simple as passing a JSON file (e.g., in format `{libfuzzer: '--flag_a --flag b'}`) as an argument of `run_experiment.py`, or having extra items in the `experiment-config.yaml` (e.g., in format `extra_flags: -libfuzzer: '--flag_a --flag b'`)?
> 
> We will also make sure to document these flags in the experiment report, though.

This seems complex to me. And lets say you are comparing libfuzzer --flaga and libfuzzer --flagb, what do you name them?
I think fuzzbench doesn't have a lot of people working on it so it makes sense to choose simple easy-to-maintain solutions.

**Comment by jonathanmetzman** [comment]
Related: https://github.com/google/fuzzbench/issues/1636

**Comment by Alan32Liu** [comment]
> #1636

1. Could we save logs to the experiment data bucket? In that case, we don't have to worry about exposing private logs.
2. Could we also save dispatcher logs? Currently, we need to [ssh to the VM to access it](https://github.com/google/fuzzbench/issues/1806#issuecomment-1496639333).

**Comment by jonathanmetzman** [comment]
>Could we save logs to the experiment data bucket? In that case, we don't have to worry about exposing private logs.

I don't think we can save it to the experiment bucket but it is being saved right now to a different bucket gs://fuzzbench-logs/

However, I haven't made this bucket public because we need a way to filter private experiments. I'm not sure I like another change I had to make to make this work https://github.com/google/fuzzbench/pull/1638/files

>Could we also save dispatcher logs? Currently, we need to https://github.com/google/fuzzbench/issues/1806#issuecomment-1496639333.

Maybe, can you think of a way to do this?

**Comment by Alan32Liu** [comment]
> However, I haven't made this bucket public because we need a way to filter private experiments. 

I see, could we have a `private-logs` bucket just like private experiment data and reports?

> > Could we also save dispatcher logs? Currently, we need to https://github.com/google/fuzzbench/issues/1806#issuecomment-1496639333.

> Maybe, can you think of a way to do this?

I noticed build logs are in the experiment data bucket. Would it be OK for me to create another directory to save dispatcher log file?

**Comment by jonathanmetzman** [comment]
> > However, I haven't made this bucket public because we need a way to filter private experiments.
> 
> I see, could we have a `private-logs` bucket just like private experiment data and reports?

I think there's a misunderstanding here. We're using a GCS router to route logs to the bucket and we must not route logs from a private experiment into a public bucket.

> 
> > > Could we also save dispatcher logs? Currently, we need to [#1806 (comment)](https://github.com/google/fuzzbench/issues/1806#issuecomment-1496639333).
> 
> > Maybe, can you think of a way to do this?
> 
> I noticed build logs are in the experiment data bucket. Would it be OK for me to create another directory to save dispatcher log file?

absolutely. I think it's OK even to put it in the top level folder.
